modules/monitoring/default.nix

```diff
@@ -147,7 +150,13 @@ let
     ];
   };
 
-  mountsFileSystemType = fsType: {} != filterAttrs (n: v: v.fsType == fsType) config.fileSystems;
+  machineMountsFileSystemType = config: fsType: {} != filterAttrs (n: v: v.fsType == fsType) config.fileSystems;
+
+  machineHasFileSystemZfs = config: mountsFileSystemType config "zfs" || elem "zfs" config.boot.supportedFilesystems;
```

Comment:
Does this line need ` || elem "zfs" config.boot.initrd.supportedFilesystems` appended?

Reply:
The latter lists filesystems that must be available in the inital ramdisk already, that's not relevant for this case.

Reply:
Btw, do we have a case we mount a zfs filesystem and don't have zfs in `supportedFilesystems`? I think that's not even possible because the option ensures that the zfs kernel module is actually loaded. If that's correct, we'd only need the second condition, i.e. `elem "zfs" config.boot.supportedFilesystems`.

Reply:
Well, indeed it is extracted like `boot.supportedFilesystems = map (fs: fs.fsType) fileSystems;` in https://github.com/mayflower/nixpkgs/blob/mf-stable/nixos/modules/tasks/filesystems.nix#L300

Hm, the double check was introduced in https://github.com/mayflower/nixexprs/commit/2bd66065873eb2fc3faeb460fb8bb1436db4d7c9#diff-06bdb0be92c5ba257a53478644db93a20035cb54f94555a95357c88be66e9481R311

But now i think we don't even need the `supportedFilesystems` check at all, i imagine we are only interested in metrics of actually mounted `zfs` filesystems, right?